### PR TITLE
Use soundtouch to process vocal audio time stretching

### PR DESCRIPTION
**Pull request by 1123852253**

### CODE REVIEW

**Comment by pauldavisthefirst** [comment]
Overall nice work. 

However, I don't think we can accept libsoundtouch becoming a mandatory dependency. So you'll need to add appropriate conditional building and use in the top level and libs/ardour wscripts and anywhere else that might need it.

File: gtk2_ardour/editor_timefx.cc [hunk 1]
```diff
@@ -218,7 +218,7 @@ Editor::time_fx (RegionList& regions, float val, bool pitching)
 
 	txt = current_timefx->stretch_opts_selector.get_active_text ();
 
-	for (int i = 0; i <= 6; i++) {
+	for (size_t i = 0; i < rb_opt_strings.size(); i++) {
```
Comment (x42) [inline gtk2_ardour/editor_timefx.cc:221]: Nice!

File: gtk2_ardour/editor_timefx.cc [hunk 2]
```diff
@@ -229,23 +229,23 @@ Editor::time_fx (RegionList& regions, float val, bool pitching)
...
+		*
+		* Actually overhaul this the dialog and processing opts below and use rubberband's
+		* "Crispness" levels:
+		*   -c 0   equivalent to --no-transients --no-lamination --window-long
+		*   -c 1   equivalent to --detector-soft --no-lamination --window-long (for piano)
+		*   -c 2   equivalent to --no-transients --no-lamination
+		*   -c 3   equivalent to --no-transients
+		*   -c 4   equivalent to --bl-transients
+		*   -c 5   default processing options
+		*   -c 6   equivalent to --no-lamination --window-short (may be good for drums)
```
[the remark this answers is not in the document]
Comment (1123852253) [reply gtk2_ardour/editor_timefx.cc:247]: The original comment is correct. This is a mistake You can restore it

**Comment by x42** [comment]
Rebased, cleaned and merged as 262281bc1f13d9 and 353ac8e106ca23e.

Thank you!

**Pull request closed without merging**